File: samples/CSharp/Northwind/Category.cs
using System;
using System.Collections.ObjectModel;
using DbExtensions;

namespace Samples.CSharp.Northwind {

   [Table(Name = "Categories")]
   public class Category {

      [Column(IsPrimaryKey = true, IsDbGenerated = true)]
      public int CategoryID { get; set; }

      [Column]
      public string CategoryName { get; set; }

      [Column]
      public string Description { get; set; }

      [Column]
      public byte[] Picture { get; set; }

      [Association(OtherKey = nameof(Product.CategoryID))]
      public Collection<Product> Products { get; } = new Collection<Product>();
   }
}

File: samples/VisualBasic/Northwind/Category.vb
Imports System
Imports System.Collections.ObjectModel
Imports DbExtensions

Namespace Northwind

   <Table(Name:="Categories")>
   Public Class Category

      <Column(IsPrimaryKey:=True, IsDbGenerated:=True)>
      Property CategoryID As Integer

      <Column>
      Property CategoryName As String

      <Column>
      Property Description As String

      <Column>
      Property Picture As Byte()

      <Association(OtherKey:=NameOf(Product.CategoryID))>
      ReadOnly Property Products As New Collection(Of Product)

   End Class

End Namespace
